lbz/websocket/request.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from multidict import CIMultiDict
+
+from lbz._request import _Request
+from lbz.authentication import User
+from lbz.misc import get_logger
+from lbz.websocket.enums import ActionType
+
+logger = get_logger(__name__)
+
+
+class WebSocketRequest(_Request):
+    """Represents request from Web Socket Secure API Gateway.
+
+    rel: https://docs.aws.amazon.com/apigateway/latest/developerguide/
+    apigateway-websocket-api-mapping-template-reference.html
+    """
+
+    def __init__(
+        self,
+        body: str | bytes | dict,
+        request_details: dict,
+        context: dict,
+        is_base64_encoded: bool,
+        user: User | None = None,
+        headers: CIMultiDict | None = None,
+    ):
+        self.headers = headers
+        self.context = context
+        self.user = user
+        self.action = request_details.pop("routeKey")
+        self.action_type = request_details.pop("eventType")
+        self.connection_id = request_details.pop("connectionId")
+        self.domain = request_details.pop("domainName")
+        self.stage = request_details.pop("stage")
+        self.details = request_details
+        super().__init__(body=body, is_base64_encoded=is_base64_encoded)
```

Comment:
The same here, it should be done as a first operation:
```suggestion
        super().__init__(body=body, is_base64_encoded=is_base64_encoded)
        self.headers = headers
        self.context = context
        self.user = user
        self.action = request_details.pop("routeKey")
        self.action_type = request_details.pop("eventType")
        self.connection_id = request_details.pop("connectionId")
        self.domain = request_details.pop("domainName")
        self.stage = request_details.pop("stage")
        self.details = request_details
```